src/standalone/Snow/Snow.jl

```diff
@@ -346,42 +353,56 @@ function ClimaLand.make_update_boundary_fluxes(model::SnowModel{FT}) where {FT}
         # First compute the boundary fluxes
         snow_boundary_fluxes!(model.boundary_conditions, model, Y, p, t)
         # Next, clip them in case the snow will melt in this timestep
-        @. p.snow.applied_water_flux = clip_total_snow_water_flux(
+        @. p.snow.applied_water_flux = clip_water_flux(
             Y.snow.S,
             p.snow.total_water_flux,
             model.parameters.Δt,
         )
+
         @. p.snow.applied_energy_flux = clip_total_snow_energy_flux(
             Y.snow.U,
             Y.snow.S,
             p.snow.total_energy_flux,
             p.snow.total_water_flux,
             model.parameters.Δt,
         )
+        # We now estimate the phase change flux: if the applied energy flux is such that T > T_f on the next step, use the residual after warming to T_f to melt snow
+        # This estimate uses the current SWE and S_l.
+        @. p.snow.phase_change_flux = phase_change_flux(
+            Y.snow.U,
+            Y.snow.S,
+            Y.snow.S_l,
+            p.snow.applied_energy_flux,
+            model.parameters,
+        )
+        @. p.snow.liquid_water_flux +=
+            p.snow.phase_change_flux * p.snow.snow_cover_fraction
+        @. p.snow.liquid_water_flux = clip_water_flux(
```

Comment:
I didnt store this an an "applied liquid water flux" because we dont need it anywhere else. but, the difference between applied and total energy/water fluxes are required to conserved energy and mass.

Reply:
The desired behavior: if there is enough heat to melt all remaining snow, S_l -> S, and then the "runoff" term damps S_l to zero.

That said, I have not clipped S_l directly (it's hard to guarantee because we can only clip tendencies of prognostic variables, not the prognostic variables themselves) but we do clip q_l to be <=1, we clipe S_l <=S in any diagnostic function using it.

Reply:
I changed it to this: 
```
function clip_liquid_water_flux(S_l::FT, S::FT, liquid_water_flux::FT, applied_water_flux::FT, Δt::FT) where {FT}
    predicted_S = S - applied_water_flux * Δt
    predicted_S_l = S_l - liquid_water_flux * Δt
    if predicted_S_l < 0
        return S_l / Δt
    elseif predicted_S_l > predicted_S
        return (S_l - predicted_S)/Δt
    else
        return liquid_water_flux
    end
end
```